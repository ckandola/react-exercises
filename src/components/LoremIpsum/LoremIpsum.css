/*
=============== 
Variables
===============
*/

:root {
    --clr-primary-1: hsl(205, 86%, 17%);
    --clr-primary-5: hsl(205, 78%, 60%);
    --clr-primary-8: hsl(205, 86%, 81%);
    --clr-primary-10: hsl(205, 100%, 96%);
    --transition: all 0.3s linear;
    --spacing: 0.1rem;
    --radius: 0.25rem;
  }
  
  .ipsum-btn {
    text-transform: uppercase;
    background: var(--clr-primary-5);
    color: var(--clr-primary-1);
    padding: 0.375rem 0.75rem;
    letter-spacing: 1px;
    display: inline-block;
    transition: var(--transition);
    font-size: 0.875rem;
    border: 2px solid var(--clr-primary-5);
    cursor: pointer;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.2);
    border-radius: var(--radius);
  }
  .ipsum-btn:hover {
    color: var(--clr-primary-5);
    background: var(--clr-primary-8);
    border-color: var(--clr-primary-8);
  }
  /* section */
  .ipsum-section {
    padding: 5rem 0;
  }
  
  .ipsum-section-center {
    width: 90vw;
    margin: 0 auto;
    max-width: 40rem;
    margin-top: 5rem;
    text-align: center;
  }
  @media screen and (min-width: 992px) {
    .ipsum-section-center {
      width: 95vw;
    }
  }
  
  /*
  =============== 
  Lorem Ipsum
  ===============
  */
  .ipsum-section-center h3 {
    text-transform: uppercase;
    color: var(--clr-primary-1);
  }
  .lorem-form {
    text-transform: capitalize;
    letter-spacing: var(--spacing);
    margin-top: 2rem;
    margin-bottom: 4rem;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  label {
    font-size: 1.25rem;
    color: var(--clr-primary-1);
  }
  .lorem-input {
    padding: 0.25rem 0.5rem;
    width: 4rem;
    border-radius: var(--radius);
    border: none;
    margin: 0 0.5rem;
    font-size: 1.25rem;
  }
  button {
    background: var(--clr-primary-10);
  }
  .ipsum-result {
    margin-bottom: 2rem;
  }
