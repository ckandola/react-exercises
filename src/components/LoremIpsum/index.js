// from https://youtu.be/a_7Z7C_JCyo?t=10352

import React, {useState} from 'react';
import './LoremIpsum.css';
import data from './data';

const LoremIpsum = () => {
    const [count, setCount] = useState(0);
    const [text, setText] = useState([]);

    const handleSubmit = e => {
        e.preventDefault();
        let amount = parseInt(count);
        amount = amount <= 0 ? 0 : amount > 8 ? 8 : amount;
        setText(data.slice(0, amount));
    }

    const handleChange = e => {
        setCount(e.target.value);
    }

    return (
        <div>
            <section className="ipsum-section-center">
                <h3>Need Lorem Ipsum?</h3>
                <form className="lorem-form" onSubmit={handleSubmit}>
                    <label htmlFor="amount">
                        paragraphs: 
                    </label>
                    <input type="number" name="amount" id="amount" value={count}
                        onChange={handleChange}/>
                    <button type="submit" className="ipsum-btn">Generate</button> 
                </form>
                <div className="lorem-text">
                    {text.map((sentence, index) => {
                        return <p key={index}>{sentence}</p>
                    })};
                </div>
            </section>
        </div>
    );
};

export default LoremIpsum;
