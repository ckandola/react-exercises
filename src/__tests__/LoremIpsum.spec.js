import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoremIpsum from '../components/LoremIpsum';

describe('LoremIpsum tests', () => {

    const wrapper = shallow(<LoremIpsum />);

    it('matches the snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders 2 paragraphs when 2 is inputted and GENERATE is clicked', () => {
        wrapper.find('.lorem-input').simulate('change', {target: {value: 2}});
        wrapper.find('form').simulate('submit', { preventDefault () {} });
        expect(wrapper.find('p')).toHaveLength(2);
    });

    it('renders at most 8 paragraphs when the input is `8` or larger and GENERATE is clicked', () => {
        wrapper.find('.lorem-input').simulate('change', {target: {value: 8}});
        wrapper.find('form').simulate('submit', { preventDefault () {} });
        expect(wrapper.find('p')).toHaveLength(8);
        wrapper.find('.lorem-input').simulate('change', {target: {value: 9}});
        wrapper.find('form').simulate('submit', { preventDefault () {} });
        expect(wrapper.find('p')).toHaveLength(8);
    });

    it('renders no paragraphs when the input is `0` or lower', () => {
        wrapper.find('.lorem-input').simulate('change', {target: {value: 0}});
        wrapper.find('form').simulate('submit', { preventDefault () {} });
        expect(wrapper.find('p')).toHaveLength(0);
        wrapper.find('.lorem-input').simulate('change', {target: {value: -2}});
        wrapper.find('form').simulate('submit', { preventDefault () {} });
        expect(wrapper.find('p')).toHaveLength(0);
    });

});
