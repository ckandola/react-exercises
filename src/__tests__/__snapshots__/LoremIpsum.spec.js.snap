// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoremIpsum tests matches the snapshot 1`] = `
<div>
  <section
    className="ipsum-section-center"
  >
    <h3>
      Need Lorem Ipsum?
    </h3>
    <form
      className="lorem-form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="amount"
      >
        paragraphs:
      </label>
      <input
        className="lorem-input"
        id="amount"
        name="amount"
        onChange={[Function]}
        type="number"
        value={0}
      />
      <button
        className="ipsum-btn"
        type="submit"
      >
        Generate
      </button>
    </form>
    <div
      className="lorem-text"
    />
  </section>
</div>
`;
